src/main/scala/chisel3/internal/firrtl/Emitter.scala

```diff
@@ -82,7 +82,7 @@ private class Emitter(circuit: Circuit) {
     */
   private def emit(m: Component): String = {
     // Generate the body.
-    val moduleName = m.id.getClass.getName.split('.').last
+    val moduleName = m.id.desiredName
```

Comment:
Yeah, this is no more incorrect than it used to be, but it's not actually that simple.  Using the legalized name breaks deduplication.  I'll see if I can fix it.